org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/formatter/FormatterRegressionTests.java

```diff
@@ -16223,4 +16223,46 @@ public void testBug564052() {
 		"		C3 {\n" +
 		"}");
 }
+
+/**
+ * https://github.com/eclipse-jdt/eclipse.jdt/issues/59 - Formatter support for guard expressions
+ */
+public void testGH59() {
+	setComplianceLevel(CompilerOptions.VERSION_21);
+
+	String source =
+		"""
+		public class JunkClass {
+		    Object myObj;
+
+		    void foo() {
+		            when(myObj);
+		        foo(myObj);
+		        foo(myObj);
+		    }
+
+		    void foo( Object stuff ) {}
+		    void when( Object methodCall ) {}
+		}

```

Comment:
It seems here you introduced a bogus `\r\n` lineend, which then escalated in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1566/files#r1395990810 (_github takes long time to actually show that comment due to size of the change_).

Reminds me that mixed line endings are dangerous :)